include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -149,17 +152,15 @@ inline void StatsdClient::send(const std::string& key,
         return;
     }
 
-    // Prepare the buffer and include the sampling rate if it's not 1.f
-    std::string buffer(256, '\0');
-    int string_len = -1;
+    // Sampling rate is 1.0f, no need to specify it
+    int string_len;
     if (isFrequencyOne) {
-        // Sampling rate is 1.0f, no need to specify it
         string_len = std::snprintf(
-            &buffer.front(), buffer.size(), "%s%s:%d|%s", m_prefix.c_str(), key.c_str(), value, type.c_str());
-    } else {
-        // Sampling rate is different from 1.0f, hence specify it
-        string_len = std::snprintf(&buffer.front(),
-                                   buffer.size(),
+            &m_buffer.front(), m_buffer.size(), "%s%s:%d|%s", m_prefix.c_str(), key.c_str(), value, type.c_str());
+    }  // Sampling rate is different from 1.0f, hence specify it
+    else {
+        string_len = std::snprintf(&m_buffer.front(),
+                                   m_buffer.size(),
```

Comment:
this is hard to read because i moved the comments above the control flow statements, but all i really did here is change `buffer` to `m_buffer`